### PR TITLE
WIP: Dutchx

### PR DESCRIPTION
**Pull request by tibetsprague**
Related issue: #6000

Tabs work, forms mostly in place, doesnt submit correctly, still need to work on whitelisting tokens

### CODE REVIEW

File: src/genericSchemeRegistry/dxDao.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  "actions": [
+    {
+      "id": "updateMasterCopy",
+      "label": "Update MasterCopy",
+      "notes": "https://github.com/gnosis/dx-contracts/blob/master/contracts/base/DxUpgrade.sol#L17",
+      "abi": {
+        "constant": false,
+        "inputs": [
+          {
+            "label": "New Contract Address",
```
Comment (jellegerbrandy) [inline src/genericSchemeRegistry/dxDao.json]: sorry we cannot do this. The ABi is a strict format, we need to save the input labels apart from that...

Comment (jellegerbrandy) [reply src/genericSchemeRegistry/dxDao.json]: something like `action.fields = { 
   _masterCopy: { label: "the label" }}
}`

File: src/components/CreateProposal/CreateProposalContainer.tsx [hunk 2]
```diff
@@ -50,8 +51,8 @@ class CreateProposalContainer extends React.Component<IProps, null> {
           : schemeName === "SchemeRegistrar" ?
             <CreateSchemeRegistrarProposal daoAvatarAddress={daoAvatarAddress} handleClose={this.goBack.bind(this)} />
           : schemeName === "GenericScheme" ?
-            <CreateGenericSchemeProposal daoAvatarAddress={daoAvatarAddress} handleClose={this.goBack.bind(this)} />
-          : ""
+            <CreateDutchXProposal daoAvatarAddress={daoAvatarAddress} handleClose={this.goBack.bind(this)} />
```
Comment (jellegerbrandy) [inline src/components/CreateProposal/CreateProposalContainer.tsx]: Given that we are tryuing to implement a generic solution here, perhaps we should rename these components and remove any references to DutchX...

**Comment by A-Zak** [comment]
A few comments:
• the arrow should be the same size as the tab :)
image
• placeholders should be a little less abi-ish - [a bunch of small changes](https://docs.google.com/spreadsheets/d/1FKjKj18MjhlOK5GapMguyOfZZ6wtWp9WcKTZu_YtK6w/edit#gid=0)
• Typo in "New auction threshold" tab title... should be "auction", not "action"
• When adding / removing multiple tokens we only need the [+] & [-] buttons once (at the bottom of the list)...
image

-----

Just to make sure, that the title of the popup will be the title of the scheme... and when it'll be called DutchX, it will also show up in the popup title...
image

**Comment by tibetsprague** [comment]
@A-Zak for the title when you say the title of "the scheme",  do you mean "DutchExchange"? because thats the name of the scheme. Or do you want it to be the name of the DAO?

**Comment by tibetsprague** [comment]
@A-Zak also i dont totally agree with the plus and minus for tokens only being at the end, its kinda handy to be able to remove one in the middle. maybe overkill, but we already have it :)

**Comment by tibetsprague** [comment]
also, i really dont think it looks good to have a field's label and placeholder be the same text. its just duplicate. the placeholder should be an example of what a correct contract address looksl ike, for example, or nothing, we may not need placeholders

**Comment by tibetsprague** [comment]
actually @A-Zak i think it just dont know what you mean by New Title in that spreadsheet. i assumed you meant the new label for the field. but also you dont mention the Token Address field

**Comment by A-Zak** [comment]
@tibetsprague 

--------

> @A-Zak for the title when you say the title of "the scheme", do you mean "DutchExchange"? because that's the name of the scheme. Or do you want it to be the name of the DAO?

I mean the "DutchExhcnage" (I thought it would be called DutchX)...

--------

> @A-Zak also i dont totally agree with the plus and minus for tokens only being at the end, its kinda handy to be able to remove one in the middle. maybe overkill, but we already have it :)

I think that in this case, it'll be more distracting than useful... let's remove them please :)

----------

> also, i really dont think it looks good to have a field's label and placeholder be the same text. its just duplicate. the placeholder should be an example of what a correct contract address looksl ike, for example, or nothing, we may not need placeholders

You're right... let's go with no placeholders in this case... When we show this to gnosis, maybe they'll have an idea for a good placeholder... but for now, let's remove them :)

---------

> actually @A-Zak i think it just dont know what you mean by New Title in that spreadsheet. i assumed you meant the new label for the field. but also you dont mention the Token Address field

Yes, I meant the field's label. the "Token address" seemed fine, so I didn't include it... Sorry for the confusion... ^_^

**Comment by tibetsprague** [comment]
@A-Zak actually we can call it whatever we want. i will go with DutchX for now

**Comment by tibetsprague** [comment]
@A-Zak you want the SchemeCard to say DutchX too? so it should never say GenericScheme if we know the specific scheme its pointing to?

**Comment by tibetsprague** [comment]
@A-Zak check out the latest, i made it so you can still remove any token in the list but its much cleaner

**Comment by A-Zak** [comment]
> @A-Zak you want the SchemeCard to say DutchX too? so it should never say GenericScheme if we know the specific scheme its pointing to?

yes, exactly :)

**Comment by A-Zak** [comment]
@tibetsprague 

> @A-Zak check out the latest, i made it so you can still remove any token in the list but its much cleaner

This looks great... we just need to start with the first field open :)

**Comment by tibetsprague** [comment]
ok done. @A-Zak let me know if you are ok with our attempts to fix the tab selected arrow. its now centered. its much tricker to try and get it to be the height of the tab, but if you want that ill get back to work on it :)

**Comment by dkent600** [comment]
@tibetsprague 

> Related issue: #6000

I think you meant 600.

Can we refer to this as follows:

> Resolves: #600 

Then the ticket will be automatically closed.

Also, can we have PR descriptions that describe what functionality changes are being proposed?

**Pull request merged**